File: backend/src/main/java/com/routinesystem/backend/services/RoutineExerciseService.java
package com.routinesystem.backend.services;

import com.routinesystem.backend.dtos.RoutineExerciseDto;
import java.util.List;

public interface RoutineExerciseService {
    RoutineExerciseDto createRoutineExercise(RoutineExerciseDto routineExerciseDto);
    RoutineExerciseDto getRoutineExerciseById(Long routineExerciseId);
    List<RoutineExerciseDto> getAllRoutineExercisesByRoutine(Long routineId);
    RoutineExerciseDto updateRoutineExercise(Long routineExerciseId, RoutineExerciseDto routineExerciseDto);
    void deleteRoutineExercise(Long routineExerciseId);
}

File: frontend/src/services/RoutineExerciseService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/routine-exercises";

class RoutineExerciseService {

    getAllRoutinesExercisesByRoutine(routineId) {
        return axios.get(`${BASE_URL}/routine/${routineId}`);
    }

    createRoutineExercise(routineExercise){
        return axios.post(BASE_URL, routineExercise);
    }

    getRoutineExerciseById(routineExerciseId){
        return axios.get(`${BASE_URL}/${routineExerciseId}`);
    }

    updateRoutineExercise(routineExerciseId, routineExercise){
        return axios.put(`${BASE_URL}/${routineExerciseId}`, routineExercise);
    }

    deleteRoutineExercise(routineExerciseId){
        return axios.delete(`${BASE_URL}/${routineExerciseId}`);
    }

}

export default new RoutineExerciseService();
